src/components/common/Tooltip/TooltipMenu.jsx

```diff
@@ -0,0 +1,105 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import Tooltip from './Tooltip';
+
+class TooltipMenu extends Component {
+  constructor(props) {
+    super(props);
+
+    this.mounted = false;
+    this.state = {
+      showPopup: false,
+    };
+
+    this.setRef = this.setRef.bind(this);
+    this.togglePopup = this.togglePopup.bind(this);
+    this.closePopup = this.closePopup.bind(this);
+    this.handleOutsideClick = this.handleOutsideClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+    window.addEventListener('scroll', this.closePopup, true);
+    window.addEventListener('mousedown', this.handleOutsideClick, true);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.closePopup);
+    window.removeEventListener('mousedown', this.handleOutsideClick);
+    this.mounted = false;
+  }
+
+  setRef(what) {
+    return (element) => { this[what] = element; };
+  }
+
+  get position() {
+    if (!this.wrapper) return { top: '0px', left: '0px' };
+    const { top, right } = this.wrapper.getBoundingClientRect();
+    return { top: `${top}px`, left: `${right}px` };
+  }
+
+  togglePopup(evt) {
+    if (this.mounted) {
+      this.setState({ showPopup: !this.state.showPopup });
```

Comment:
I wonder why the set state has to be guarded with the mounted bool? It doesn't seem like it would be sensitive to mounting (same below)

Reply:
Good question. Don't remember why I did this; might be from copying some other component. Will try removing it